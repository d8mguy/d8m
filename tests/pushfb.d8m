// small test for combining pushf and pushb
val prtup = tuple(a, b: integer)
val foo = \imp() {
    var lst1: list(integer) = []
    var lst2: list(prtup) = []
    each(i^1..6) {
        val ptp = [prtup: i, i*2]
        if(i%2 == 0) lst1.pushf(i) else lst1.pushb(i)
        if(i%2 == 0) lst2.pushf(ptp) else lst2.pushb(ptp)
    }
    [a~lst1, b~lst2]
}
foo()
