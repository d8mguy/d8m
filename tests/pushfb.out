[[6, 4, 2, 1, 3, 5]; [[6; 12], [4; 8], [2; 4], [1; 2], [3; 6], [5; 10]]]
